src/lib/spacefinder/article-body-adverts.ts

```diff
@@ -200,84 +233,56 @@ const addDesktopInlineAds = (isInline1: boolean): Promise<boolean> => {
 		filter: filterNearbyCandidates(adSizes.halfPage.height),
 	};
 
-	const rules = isInline1 ? firstInlineRules : subsequentInlineRules;
-
 	const insertAds: SpacefinderWriter = async (paras) => {
-		// Make ads sticky on the non-inline1 pass
-		// i.e. inline2, inline3, etc...
-		const isSticky = !isInline1;
-
-		if (isSticky) {
-			const stickyContainerHeights = await computeStickyHeights(
-				paras,
-				articleBodySelector,
-			);
+		const stickyContainerHeights = await computeStickyHeights(
+			paras,
+			articleBodySelector,
+		);
 
-			void insertHeightStyles(
-				stickyContainerHeights.map((height, index) => [
-					getStickyContainerClassname(index),
-					height,
-				]),
-			);
-		}
+		void insertHeightStyles(
+			stickyContainerHeights.map((height, index) => [
+				getStickyContainerClassname(index),
+				height,
+			]),
+		);
+
+		const slots = paras.slice(0, paras.length).map(async (para, i) => {
+			const isLastInline = i === paras.length - 1;
+
+			const containerClasses =
+				getStickyContainerClassname(i) +
+				' offset-right ad-slot--offset-right ad-slot-container--offset-right';
+
+			const containerOptions = {
+				sticky: true,
+				className: containerClasses,
+			};
 
-		const slots = paras
-			.slice(0, isInline1 ? 1 : paras.length)
-			.map(async (para, i) => {
-				const inlineId = i + (isInline1 ? 1 : 2);
-				const isLastInline = i === paras.length - 1;
-
-				let containerClasses = '';
-
-				if (isSticky) {
-					containerClasses += getStickyContainerClassname(i);
-				}
-
-				if (!isInline1) {
-					containerClasses +=
-						' offset-right ad-slot--offset-right ad-slot-container--offset-right';
-				}
-
-				const containerOptions = {
-					sticky: isSticky,
-					className: containerClasses,
-				};
-
-				return insertAdAtPara(
-					para,
-					`inline${inlineId}`,
-					'inline',
-					'inline',
-					// these are added here and not in size mappings because the inline[i] name is also used on fronts, where we don't want outstream or tall ads
-					isInline1
-						? {
-								phablet: [
-									adSizes.outstreamDesktop,
-									adSizes.outstreamGoogleDesktop,
-								],
-								desktop: [
-									adSizes.outstreamDesktop,
-									adSizes.outstreamGoogleDesktop,
-								],
-						  }
-						: {
-								desktop: await decideAdditionalSizes(
-									para,
-									[adSizes.halfPage, adSizes.skyscraper],
-									isLastInline,
-								),
-						  },
-					containerOptions,
-				);
-			});
+			return insertAdAtPara(
+				para,
+				`inline${i + 2}`,
+				'inline',
+				'inline',
+				// these are added here and not in size mappings because the inline[i] name
+				// is also used on fronts, where we don't want outstream or tall ads
+				{
+					desktop: await decideAdditionalSizes(
+						para,
+						[adSizes.halfPage, adSizes.skyscraper],
+						isLastInline,
+					),
+				},
```

Comment:
Take it or leave it, but I really like out how you pulled the `additionalSizes` in the first pass out into a constant, perhaps we could do the same here?

```typescript
const additionalSizes = {
	desktop: await decideAdditionalSizes(
		para,
		[adSizes.halfPage, adSizes.skyscraper],
		isLastInline,
	),
};
```